lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2774,8 +2774,23 @@ std::string SPIRVToLLVM::getOCLBuiltinName(SPIRVInstruction *BI) {
       return OCLSPIRVBuiltinMap::rmap(OC);
     }
     if (DataTy) {
-      if (DataTy->getBitWidth() == 16)
+      switch (DataTy->getBitWidth()) {
+      case 8:
+        Name << "_uc";
+        break;
+      case 16:
         Name << "_us";
+        break;
+      case 32:
+        // Intentionally does nothing since _ui variant is only an alias.
+        break;
+      case 64:
+        Name << "_ul";
+        break;
+      default:
```

Comment:
There are 2 very similar pieces of code. Can we move it to an util function? Will it work if we will return "_ui" in the reader?

Reply:
I think that it is better to not return "_ui" because cl_intel_subgroups uses version only without suffix. Alias with suffix was added in subsequent extensions (char, short, long). However, I can try to drop "_ui" from "visitSubgroupBlockReadINTEL". As I understand, in this function some kind of demangling is done and postfix is added only to distinguish between different signatures. Without "_ui" all variants will be different too and everying should work fine.
Will it be fine if I move code for postfixes to OCLUtil?

Reply:
> Will it be fine if I move code for postfixes to OCLUtil?

Yes